src/CHIPBackend.hh

```diff
@@ -1836,19 +1842,22 @@ public:
   virtual ~CHIPQueue();
 
   CHIPQueueFlags getQueueFlags() { return QueueFlags_; }
-  virtual void updateLastEvent(CHIPEvent *ChipEv) {
-    if (LastEvent_ != nullptr)
+  virtual void updateLastEvent(CHIPEvent *NewEvent) {
+    if (NewEvent == LastEvent_)
+      return;
+
+    if (LastEvent_ != nullptr) {
       std::lock_guard<std::mutex> LockLast(LastEvent_->Mtx);
+      LastEvent_->decreaseRefCount("updateLastEvent - old event");
```

Comment:
The mutex _is_ part of the `ChipEvent` class. The reason why it's not locking inside `decreaseRefCount()` is because that can cause a hang with callbacks. Thread safety should be improved and I will address it in a separate PR.